src/utils/fetchTautulli.ts

```diff
@@ -90,23 +93,64 @@ export async function getLibraries(
   excludeInactive = true,
 ): Promise<TautulliLibrary[]> {
   const settings = getSettings()
+  const libraries = (await fetchTautulli<TautulliLibrary[]>('get_libraries'))
+    ?.response?.data
+  if (libraries == null) {
+    console.warn('[TAUTULLI] - No libraries found!')
+    return []
+  }
+
+  const activeLibraries = filterByActiveLibraries(
+    excludeInactive,
+    settings,
+    libraries,
+  )
+  return await filterBySharedLibraries(settings, activeLibraries)
+}
+
+function filterByActiveLibraries(
+  excludeInactive: boolean,
+  settings: Settings,
+  libraries: TautulliLibrary[],
+): TautulliLibrary[] {
   const activeLibraries = settings.general.activeLibraries
-  const libraries = await fetchTautulli<TautulliLibrary[]>('get_libraries')
+  return excludeInactive
+    ? libraries.filter((library) =>
+        activeLibraries.includes(kebabCase(library.section_name)),
+      )
+    : libraries
+}
 
-  if (!libraries) {
-    console.warn('[TAUTULLI] - No libraries found!')
+async function filterBySharedLibraries(
+  settings: Settings,
+  libraries: TautulliLibrary[],
+): Promise<TautulliLibrary[]> {
+  if (settings.general.isOutsideAccess) {
+    return libraries
+  }
+
+  const session = await getServerSession(authOptions)
+  const userId = session?.user.id
+
+  if (!userId) {
+    console.error('[TAUTULLI] - No user ID found!')
     return []
   }
 
-  if (excludeInactive) {
-    const filteredLibraries = libraries.response.data.filter((library) =>
-      activeLibraries.includes(kebabCase(library.section_name)),
-    )
+  const userDataResponse = await fetchTautulli<TautulliUser>('get_user', {
+    user_id: userId,
+  })
 
-    return filteredLibraries
-  } else {
-    return libraries.response.data
+  if (!userDataResponse) {
+    console.warn(`[TAUTULLI] - Could not fetch user data for user ID ${userId}`)
+    return []
   }
+
+  return libraries.filter((library) =>
+    userDataResponse.response.data.shared_libraries.includes(
```

Comment:
I noticed you didn't add conditional checks to `response.data` here, but did above in

`const libraries = (await fetchTautulli<TautulliLibrary[]>('get_libraries'))?.response?.data`

There doesn't actually seem to be consistency regarding this in the codebase currently. Seems to be done both ways. Let me know your thoughts regarding which we should prefer. Personally, I'm thinking the extra conditional wouldn't hurt and would just add extra safety. Eitherway, would be good to streamline this.

Reply:
I think `get_libraries` one should drop the checks. I think we be should using `?` with caution. It can lead to hacky code. Generelly I think `fetchTautulli` should be refactoredd to not return null values and return `data` of `response `directly. Intead we should throw errors and handle error cases that way. This would propably affect a big chunk of the code base